Makefile

```diff
@@ -1,64 +1,111 @@
+# Convenience data so we can run the following and include
+# sources from three directories deep.
+#
+# Example:
+# $(subst dir, emily/cdk/lib, $(THREE_DIRS_DEEP))
+# becomes
+# emily/cdk/lib/*  emily/cdk/lib/*/*  emily/cdk/lib/*/*/*
+THREE_DIRS_DEEP := dir/* dir/*/* dir/*/*/*
+TWO_DIRS_DEEP := dir/* dir/*/*
+
+AUTOGENERATED_BLOCKLIST_CLIENT_CLIENT := .generated-sources/blocklist-api/src/lib.rs
+BLOCKLIST_OPENAPI_PATH=./.generated-sources/blocklist-openapi-gen
+BLOCKLIST_OPENAPI_SPEC=$(BLOCKLIST_OPENAPI_PATH)/blocklist-client-openapi.json
+
+AUTOGENERATED_EMILY_CLIENT := .generated-sources/emily/src/lib.rs
+EMILY_LAMBDA_BINARY := target/lambda/emily-lambda/bootstrap.zip
+EMILY_CDK_TEMPLATE := emily/cdk/cdk.out/EmilyStack.template.json
+EMILY_DOCKER_COMPOSE := docker-compose.emily.yml
+INSTALL_TARGET := pnpm-lock.yaml
+
+# Don't use the install target here so you can rerun install without
+# Makefile complaints.
 install:
 	pnpm install
+	touch pnpm-lock.yaml
 
-build: install | emily-client-source generate-openapi-spec blocklist-api-source
+build: $(INSTALL_TARGET) $(AUTOGENERATED_EMILY_CLIENT) $(AUTOGENERATED_BLOCKLIST_CLIENT_CLIENT)
 	cargo build
 	pnpm --recursive build
 
-test: install | emily-client-source blocklist-api-source
+test: $(INSTALL_TARGET) $(AUTOGENERATED_EMILY_CLIENT) $(AUTOGENERATED_BLOCKLIST_CLIENT_CLIENT)
 	cargo test
 	pnpm --recursive test
 
-integration-test: install | emily-client-source blocklist-api-source
+integration-test: $(INSTALL_TARGET) $(AUTOGENERATED_EMILY_CLIENT) $(AUTOGENERATED_BLOCKLIST_CLIENT_CLIENT)
 	docker compose --file docker-compose.test.yml up --detach
 	cargo test
 	cargo test --test '*' --all-features -- --test-threads=1
 	pnpm --recursive test
 	docker compose --file docker-compose.test.yml down
 
-lint: install | emily-client-source blocklist-api-source
+lint: $(INSTALL_TARGET) $(AUTOGENERATED_EMILY_CLIENT) $(AUTOGENERATED_BLOCKLIST_CLIENT_CLIENT)
 	cargo clippy -- -D warnings
 	pnpm --recursive run lint
 
 clean:
 	cargo clean
 	pnpm --recursive clean
+	rm -rf devenv/dynamodb/data/*
+	@touch package.json
+
+.PHONY: install build test integration-test lint clean
+
+$(INSTALL_TARGET): $(wildcard package* */package* */*/package*)
+	pnpm install
+	touch pnpm-lock.yaml
 
 # Emily API
 # ----------------------------------------------------
 
-EMILY_PATH=emily
+EMILY_DEVENV_PATH=emily/devenv
+EMILY_LAMBDA_PATH=emily/lambda
 EMILY_API_PROJECT_NAME=emily-api
 EMILY_CDK_PROJECT_NAME=emily-cdk
+CONTAINER_HOST=host.docker.internal
+
+ifeq ($(findstring Linux, $(shell uname)), Linux)
+_CONTAINER_HOST := localhost
+else
+_CONTAINER_HOST := host.docker.internal
```

Comment:
Oh interesting, I had thought the `host.docker.internal` path would work on both macOS and linux. Good to know.